### PR TITLE
Improve AbstractTuple class

### PR DESCRIPTION
**Pull request by bsonntag**
This improves the AbstractTuple class by simplifying the method implementations and by adding javadocs with `{@inheritDoc}`.
This also adds tests for `toString` methods.

### CODE REVIEW

File: jtuples/src/main/java/org/jtuples/AbstractTuple.java [hunk 1]
```diff
@@ -26,48 +26,63 @@
...
     @Override
     public final String toString() {
-        StringJoiner joiner = new StringJoiner(", ", "(", ")");
-        for (Object e: this.asList()) {
-            joiner.add(String.valueOf(e));
-        }
-        return joiner.toString();
+        return this.asList().stream()
+                .map(String::valueOf)
+                .collect(Collectors.joining(", ", "(", ")"));
```
Comment (bsonntag) [inline jtuples/src/main/java/org/jtuples/AbstractTuple.java]: @git-afsantos What do you think about this implementation?

Comment (git-afsantos) [inline jtuples/src/main/java/org/jtuples/AbstractTuple.java]: @bsonntag It is clearly more functional, but is there any advantage to it that I am missing, besides code readability?

Comment (bsonntag) [inline jtuples/src/main/java/org/jtuples/AbstractTuple.java]: Not really. On the contrary, I think it uses more memory and produces more garbage.
Which one do you prefer?

Comment (git-afsantos) [inline jtuples/src/main/java/org/jtuples/AbstractTuple.java]: Leave it as it is. We will go back and optimize if there is any reason to optimize.

**Comment by git-afsantos** [comment]
:+1:

**Pull request merged**